### ISSUE TITLE
Editor Crash on opening scene - ERROR: Index p_index = 1 is out of bounds (((Vector<T> *)(this))->_cowdata.size() = 0).    at: operator[] (./core/templates/vector.h:52)

### ISSUE THREAD

**Issue by timshannon** [godotengine/godot#61103]
### Godot version

v4.0.alpha.mono.custom_build [fa2b3269a]

### System information

Ubuntu 22.04

### Issue description

Opening a scene that opened find in previous versions now crashes the editor.

The last error before the stack trace is 
```
ERROR: Index p_index = 1 is out of bounds (((Vector<T> *)(this))->_cowdata.size() = 0).
   at: operator[] (./core/templates/vector.h:52)
```

The full terminal output is attached in out.txt.

It appears to be an issue to the CSG nodes in the scene.  If I can find a previous GD4 alpha version in which the scene opens fine, I'll include that in this issue so it's easier to find a diff.

### Steps to reproduce

Open the attached project, then open the TestEnemy.tscn scene.

### Minimal reproduction project

[Bug.zip](https://github.com/godotengine/godot/files/8703542/Bug.zip)
[out.txt](https://github.com/godotengine/godot/files/8703543/out.txt)

**Comment by timshannon** [comment]
It opens fine in Alpha 7 but not Alpha 8.

**Comment by timshannon** [comment]
I believe this PR is what is causing this specific crash for me:

https://github.com/godotengine/godot/pull/51591

The build from the PR merged prior to #51591 (https://github.com/godotengine/godot/pull/46208) works fine.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/60936.
